src/reynir/bintokenizer.py

```diff
@@ -1375,12 +1378,38 @@ def not_in_bin(token) -> bool:
         while True:
 
             if not concatable and not is_interesting(token):
-                yield token
-                # Make sure that token is None if next() raises StopIteration
-                token = cast(Tok, None)
-                token = next(token_stream)
-                continue
-
+                if token.txt and token.txt.split(" ")[-1] in FOREIGN_MIDDLE_NAME_SET and " " in token.txt:
```

Comment:
Better: `if token.txt and " " in token.txt and token.txt.split()[-1] in FOREIGN_MIDDLE_NAME_SET:`

Reply:
Done